tests/stream_test.c

```diff
@@ -46,6 +52,24 @@ static struct aws_input_stream *s_create_binary_file_stream(struct aws_allocator
     return aws_input_stream_new_from_file(allocator, s_test_file_name);
 }
 
+static struct aws_input_stream *s_create_read_only_file_stream(struct aws_allocator *allocator) {
+    remove(s_test_file_name);
+
+    FILE *file = aws_fopen(s_test_file_name, "w+b");
```

Comment:
Maybe we should do a purge of all uses "w+" and "r+" in this library. I don't think there are any actual cases where we want to do both read and write. The unnecessary "+" encourages people to keep doing that